content/en/continuous_integration/setup_tests/javascript.md

```diff
@@ -70,56 +59,69 @@ module.exports = {
 {{< /code-block >}}
 
-**Note**: The default configuration should work for most cases, but depending on the volume and speed of your tests, the tracer or the Agent might drop some of the spans. Alleviate this by increasing the `flushInterval` (a value in milliseconds) when initializing the tracer:
+Run your tests as you normally do, specifying the environment where test are being run (e.g. `local` when running tests on a developer workstation, or `ci` when running them on a CI provider) in the `DD_ENV` environment variable. For example:
 
-{{< code-block lang="javascript" >}}
-require('dd-trace').init({
-  flushInterval: 300000
-})
+{{< code-block lang="bash" >}}
+DD_ENV=ci npm test
 {{< /code-block >}}
 
 ### Mocha instrumentation
 
 Add `--require dd-trace/init` to however you normally run your `mocha` tests, for example in your `package.json`:
 
-{{< code-block lang="javascript" >}}
-// package.json
+{{< code-block lang="javascript" filename="package.json" >}}
 'scripts': {
-  'test': 'mocha --require dd-trace/init'
+  'test': 'DD_SERVICE=my-ui-app mocha --require dd-trace/init'
 },
 {{< /code-block >}}
 
-## Disabling instrumentation in local development
+Run your tests as you normally do, specifying the environment where test are being run (e.g. `local` when running tests on a developer workstation, or `ci` when running them on a CI provider) in the `DD_ENV` environment variable. For example:
 
-If you want to disable the testing instrumentation for local development (where you might not be running the Datadog Agent), these are some options:
+{{< code-block lang="bash" >}}
+DD_ENV=ci npm test
+{{< /code-block >}}
 
-### Jest
+## Additional configuration settings
 
-When initializing the tracer, check whether you are in CI:
+The following is a list of the most important configuration settings that can be used with the tracer. They can be either passed in on its `init()` function, or as environment variables:
 
-{{< code-block lang="javascript" >}}
-require('dd-trace').init({
-  enabled: !!process.env.CI // the environment variable to use depends on the CI provider
-})
-{{< /code-block >}}
+`service`
+: Name of the service or library under test.<br/>
+**Environment variable**: `DD_SERVICE`<br/>
+**Default**: `unnamed-java-app`<br/>
+**Example**: `my-java-app`
 
-### Mocha
+`env`
+: Name of the environment where tests are being run.<br/>
+**Environment variable**: `DD_ENV`<br/>
+**Default**: `(empty)`<br/>
+**Examples**: `local`, `ci`
 
-Use different test scripts for CI and local development:
+`enabled`
+: Setting this to `false` completely disables the instrumentation.<br/>
+**Environment variable**: `DD_TRACE_ENABLED`<br/>
+**Default**: `true`
+
+`hostname`
+: The Datadog Agent hostname.<br/>
+**Environment variable**: `DD_TRACE_AGENT_HOSTNAME`<br/>
+**Default**: `localhost`
+
+`port`
+: The Datadog Agent trace collection port.<br/>
+**Environment variable**: `DD_TRACE_AGENT_PORT`<br/>
+**Default**: `8126`
```

Comment:
do we want to single out these parameters?

Reply:
Yes, as in the case of running this in a SaaS provider, they would have to change them to point to the agent running in a sidecar container instead of the default `localhost`